xdsl/transforms/csl_stencil_to_csl_wrapper.py

```diff
@@ -0,0 +1,232 @@
+                },
+            )
+            routes = csl_wrapper.ImportModuleOp(
+                "routes.csl",
+                {
+                    "pattern": param_pattern,
+                    "peWidth": param_width,
+                    "peHeight": param_height,
+                },
+            )
+
+            # set up program param `stencil_comms_params`
+            all_routes = csl.MemberCallOp(
+                "computeAllRoutes",
+                csl.ComptimeStructType(),
+                routes,
+                params=[
+                    param_x,
+                    param_y,
+                    param_width,
+                    param_height,
+                    param_pattern,
+                ],
+            )
+            # set up program param `memcpy_params`
+            memcpy_params = csl.MemberCallOp(
+                "get_params",
+                csl.ComptimeStructType(),
+                memcpy,
+                params=[
+                    param_x,
+                ],
+            )
+
+            # set up program param `is_border_region_pe`
+            one = arith.Constant(IntegerAttr(1, 16))
+            pattern_minus_one = arith.Subi(one, param_pattern)
+            width_minus_x = arith.Subi(param_width, param_x)
+            height_minus_y = arith.Subi(param_height, param_y)
+            x_lt_pattern_minus_one = arith.Cmpi(param_x, pattern_minus_one, "slt")
+            y_lt_pattern_minus_one = arith.Cmpi(param_y, pattern_minus_one, "slt")
+            width_minus_one_lt_pattern = arith.Cmpi(width_minus_x, param_pattern, "slt")
+            height_minus_one_lt_pattern = arith.Cmpi(
+                height_minus_y, param_pattern, "slt"
+            )
+            or1_op = arith.OrI(x_lt_pattern_minus_one, y_lt_pattern_minus_one)
+            or2_op = arith.OrI(or1_op, width_minus_one_lt_pattern)
+            is_border_region_pe = arith.OrI(or2_op, height_minus_one_lt_pattern)
+
+            # yield things as named params to the program module
+            csl_wrapper.YieldOp.from_field_name_mapping(
+                field_name_mapping={
+                    "memcpy_params": memcpy_params.results[0],
+                    "stencil_comms_params": all_routes.results[0],
+                    "isBorderRegionPE": is_border_region_pe.result,
+                }
+            )
+
+
+@dataclass(frozen=True)
+class CslStencilToCslWrapperPass(ModulePass):
+    """
+    Wraps program in the csl_stencil dialect in a csl_wrapper by translating each
+    top-level function to one module wrapper.
+    """
+
+    name = "csl-stencil-to-csl-wrapper"
+
+    def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
+        module_pass = PatternRewriteWalker(
+            GreedyRewritePatternApplier(
+                [
+                    ConvertStencilFuncToModuleWrappedPattern(),
+                ]
+            ),
+            walk_reverse=False,
+            apply_recursively=False,
```

Comment:
Is there a reason why you specify these two parameters here?

Reply:
Just so that it's clear what it does.

Reply:
I think it's safe to drop at least `walk_reverse`, as it's the default value anyway...

Reply:
ok